controller/util/cf_v2_client.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	model "github.com/kubernetes-incubator/service-catalog/model/service_broker"
+	scmodel "github.com/kubernetes-incubator/service-catalog/model/service_controller"
+)
+
+const (
+	catalogFormatString         = "%s/v2/catalog"
+	serviceInstanceFormatString = "%s/v2/service_instances/%s"
+	bindingFormatString         = "%s/v2/service_instances/%s/service_bindings/%s"
+)
+
+type cfV2BrokerClient struct {
+	broker *scmodel.ServiceBroker
+}
+
+// Verify that client implements the BrokerClient interface.
+var _ BrokerClient = (*cfV2BrokerClient)(nil)
```

Comment:
The variant of the interface assertion pattern I'm familiar with is `var _ Interface = &Thing{}`.  Any rationale for the form you chose?  Asking because I don't know go the way I know Java.